### PR TITLE
fix(instrumentation): do not pin OTEL dependencies

### LINKED ISSUE #21473: Fixed version of `@opentelemetry/*` dependencies
### Bug description

Currently prisms locks versions of OTEL libraries:

```json
    "@opentelemetry/api": "1.6.0",
    "@opentelemetry/instrumentation": "0.43.0",
    "@opentelemetry/sdk-trace-base": "1.17.0",
```

This creates problem with hoisting: if there are other versions of OTEL libs required in the project package manager will end up installing two copies of "@opentelemetry/api" in order to satisfy constraints:
- one in root `node_modules`
- second one in `node_modules/<another_package>/node_modules`

This will make either external or internal consumers of prisma to receive no-op implementation of a tracer.

### Expected behavior

A package should keep versions of it's dependencies as open as possible https://github.com/prisma/prisma/blob/811babfebbd8c573109f0dc083634bc2e865a87e/packages/instrumentation/package.json

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

### PR DESCRIPTION
**Pull request by maxkomarychev**
Fixes issues with telemetry https://github.com/prisma/prisma/issues/21473 by using non-pinned versions of OTEL libraries.

Fixes #21473

### CODE REVIEW

File: packages/instrumentation/package.json [hunk 1]
```diff
@@ -22,9 +22,9 @@
     "typescript": "5.3.3"
   },
   "dependencies": {
-    "@opentelemetry/api": "1.8.0",
-    "@opentelemetry/instrumentation": "0.49.1",
-    "@opentelemetry/sdk-trace-base": "1.22.0"
+    "@opentelemetry/api": "^1.7",
+    "@opentelemetry/instrumentation": "0.48 || 0.49",
```
Comment (Jolg42) [inline packages/instrumentation/package.json:26]: Note that v0 behaves differently than v1
See online example 
https://jubianchi.github.io/semver-check/#/^0.48/0.49.1
Screenshot 2024-03-06 at 18 09 46

https://jubianchi.github.io/semver-check/#/0.48%20||%200.49/0.49.1
Screenshot 2024-03-06 at 18 09 29

Comment (maxkomarychev) [reply packages/instrumentation/package.json:26]: I would actually make it open too

image


what do you think?

Comment (maxkomarychev) [reply packages/instrumentation/package.json:26]: I think 0.48 appeared here only because pr is very old

Comment (maxkomarychev) [reply packages/instrumentation/package.json]: I've pushed an update to just make it `^0.49`

**Comment by maxkomarychev** [comment]
Dear prisma maintainers, could you please provide specific feedback regarding prospects of this being merged? Are you willing to accept this pr, should I keep keeping it up to date etc?

File: packages/instrumentation/package.json [hunk 2]
```diff
@@ -22,9 +22,9 @@
     "typescript": "5.3.3"
   },
   "dependencies": {
-    "@opentelemetry/api": "1.8.0",
-    "@opentelemetry/instrumentation": "0.49.1",
-    "@opentelemetry/sdk-trace-base": "1.22.0"
+    "@opentelemetry/api": "^1.8",
+    "@opentelemetry/instrumentation": "^0.49",
```
Comment (aqrln) [inline packages/instrumentation/package.json:26]:
```suggestion
    "@opentelemetry/instrumentation": "^0.49 || ^0.50",
```

**Review by aqrln — APPROVED** [review]
Sorry for the delay here — let's finally ship this

**Comment by aqrln** [comment]
I think this may also need some adjustments in the Renovate config to avoid the versions being automatically pinned.
@Jolg42 could you help with this?

File: packages/instrumentation/package.json (hunk 2 above)
Comment (maxkomarychev) [reply packages/instrumentation/package.json]: thanks!

I appreciate your feedback. I've updated version ranges to include current and previous versions:



image


Thanks!

**Comment by maxkomarychev** [comment]
thanks!

**Pull request closed without merging**